File: services/blog-service/src/main/java/com/platform/blog/domain/entity/Article.java
package com.platform.blog.domain.entity;

import com.baomidou.mybatisplus.annotation.TableName;
import com.platform.common.core.domain.entity.BaseEntity;
import lombok.Data;
import lombok.EqualsAndHashCode;

import java.time.LocalDateTime;

/**
 * 文章实体
 */
@Data
@EqualsAndHashCode(callSuper = true)
@TableName("biz_article")
public class Article extends BaseEntity {

    /**
     * 标题
     */
    private String title;

    /**
     * 摘要
     */
    private String summary;

    /**
     * 内容
     */
    private String content;

    /**
     * 封面图
     */
    private String cover;

    /**
     * 作者ID
     */
    private Long authorId;

    /**
     * 分类ID
     */
    private Long categoryId;

    /**
     * 状态: 0-草稿, 1-已发布, 2-已归档
     */
    private Integer status;

    /**
     * 浏览数
     */
    private Integer viewCount;

    /**
     * 点赞数
     */
    private Integer likeCount;

    /**
     * 收藏数
     */
    private Integer favoriteCount;

    /**
     * 评论数
     */
    private Integer commentCount;

    /**
     * 是否置顶
     */
    private Integer isTop;

    /**
     * 是否原创
     */
    private Integer isOriginal;

    /**
     * 来源URL
     */
    private String sourceUrl;

    /**
     * 发布时间
     */
    private LocalDateTime publishTime;
}

File: frontend/platform-web/src/api/article.ts
import request from '@/utils/request'
import type { ApiResult, PageResult, PageQuery } from '@/types/api'
import type { TagVO } from './tag'

/** 文章信息 */
export interface ArticleVO {
  id: number
  title: string
  summary: string
  cover: string
  authorId: number
  authorName: string
  categoryId: number
  categoryName: string
  status: number
  viewCount: number
  likeCount: number
  favoriteCount: number
  commentCount: number
  isTop: number
  isOriginal: number
  publishTime: string
  createTime: string
  tags: TagVO[]
}

/** 文章详情 */
export interface ArticleDetailVO extends ArticleVO {
  content: string
  sourceUrl: string
  tagIds: number[]
}

/** 文章查询参数 */
export interface ArticleQueryParams extends PageQuery {
  title?: string
  categoryId?: number
  tagId?: number
  status?: number
  authorId?: number
  applyDataScope?: boolean
}

/** 创建文章参数 */
export interface ArticleCreateParams {
  title: string
  summary?: string
  content?: string
  cover?: string
  categoryId?: number
  tagIds?: number[]
  isTop?: number
  isOriginal?: number
  sourceUrl?: string
}

/** 更新文章参数 */
export interface ArticleUpdateParams {
  title: string
  summary?: string
  content?: string
  cover?: string
  categoryId?: number
  tagIds?: number[]
  isTop?: number
  isOriginal?: number
  sourceUrl?: string
}

/**
 * 分页查询文章
 */
export function getArticlePage(params: ArticleQueryParams) {
  return request.get<ApiResult<PageResult<ArticleVO>>>('/blog/articles', { params })
}

/**
 * 获取文章详情
 */
export function getArticleById(id: string | number) {
  return request.get<ApiResult<ArticleDetailVO>>(`/blog/articles/${id}`)
}

/**
 * 创建文章
 */
export function createArticle(data: ArticleCreateParams) {
  return request.post<ApiResult<string>>('/blog/articles', data)
}

/**
 * 更新文章
 */
export function updateArticle(id: string | number, data: ArticleUpdateParams) {
  return request.put<ApiResult<void>>(`/blog/articles/${id}`, data)
}

/**
 * 删除文章
 */
export function deleteArticle(id: string | number) {
  return request.delete<ApiResult<void>>(`/blog/articles/${id}`)
}

/**
 * 发布文章
 */
export function publishArticle(id: string | number) {
  return request.put<ApiResult<void>>(`/blog/articles/${id}/publish`)
}

/**
 * 归档文章
 */
export function archiveArticle(id: string | number) {
  return request.put<ApiResult<void>>(`/blog/articles/${id}/archive`)
}
